src/wp-includes/class-wp-query.php

```diff
@@ -2641,10 +2642,8 @@ public function get_posts() {
 				$queried_post_types = get_post_types( array( 'exclude_from_search' => false ) );
 			} elseif ( is_array( $post_type ) ) {
 				$queried_post_types = $post_type;
-			} elseif ( ! empty( $post_type ) ) {
```

Comment:
Since we set $post_type on top and there is no setter after that so we can remove a empty check here.

Reply:
One know, negative side effect for setting `$post_type` above is, a miss match with current flow. 

For current follow if  `$this->is_page` is true, then `$queried_post_types` will be `array( 'post' )`
which seems to be wrong. @spacedmonkey  I don't have much context here, if you will current flow is correct do let me knwo. 

With current change for `$this->is_page`, `$queried_post_types` will be array( 'page' ).

same for `$this->is_attachment`

Reply:
Did further analysis and found  `is_singular` would be true for `is_page` and `is_attachement` so old flow is same as new. 

`$this->is_singular = $this->is_single || $this->is_page || $this->is_attachment;`